pkg/pipeline/write/testnorace/write_ipfix_test.go

```diff
@@ -138,51 +203,85 @@ func TestEnrichedIPFIXFlow(t *testing.T) {
+
+	// Check data
+	assert.Equal(t, uint16(10), dataMsg.GetVersion())
+	dataSet := dataMsg.GetSet()
+	record := dataSet.GetRecords()[0]
+
+	for _, name := range write.IPv4IANAFields {
+		element, _, exist := record.GetInfoElementWithValue(name)
+		assert.Truef(t, exist, "element with name %s should exist in the record", name)
+		assert.NotNil(t, element)
+		matchElement(t, element, flow)
+	}
+
+	// Make sure enriched fields are absent
+	for _, name := range write.KubeFields {
+		element, _, exist := record.GetInfoElementWithValue(name)
+		assert.Falsef(t, exist, "element with name %s should NOT exist in the record", name)
+		assert.Nil(t, element)
+	}
+}
+
 func matchElement(t *testing.T, element entities.InfoElementWithValue, flow config.GenericMap) {
-	switch element.GetName() {
-	case "sourceIPv4Address":
-		assert.Equal(t, flow["SrcAddr"], element.GetIPAddressValue().String())
-	case "destinationIPv4Address":
-		assert.Equal(t, flow["DstAddr"], element.GetIPAddressValue().String())
-	case "ethernetType":
-		assert.Equal(t, flow["Etype"], uint32(element.GetUnsigned16Value()))
-	case "flowDirection":
-		assert.Equal(t, flow["IfDirections"], []int{int(element.GetUnsigned8Value())})
-	case "protocolIdentifier":
-		assert.Equal(t, flow["Proto"], uint32(element.GetUnsigned8Value()))
-	case "sourceTransportPort":
-		assert.Equal(t, flow["SrcPort"], uint32(element.GetUnsigned16Value()))
-	case "destinationTransportPort":
-		assert.Equal(t, flow["DstPort"], uint32(element.GetUnsigned16Value()))
-	case "octetDeltaCount":
-		assert.Equal(t, flow["Bytes"], element.GetUnsigned64Value())
-	case "flowStartMilliseconds":
-		assert.Equal(t, flow["TimeFlowStartMs"], int64(element.GetUnsigned64Value()))
-	case "flowEndMilliseconds":
-		assert.Equal(t, flow["TimeFlowEndMs"], int64(element.GetUnsigned64Value()))
-	case "packetDeltaCount":
-		assert.Equal(t, flow["Packets"], element.GetUnsigned64Value())
-	case "interfaceName":
-		assert.Equal(t, flow["Interfaces"], []string{element.GetStringValue()})
-	case "sourcePodNamespace":
-		assert.Equal(t, flow["SrcK8S_Namespace"], element.GetStringValue())
-	case "sourcePodName":
-		assert.Equal(t, flow["SrcK8S_Name"], element.GetStringValue())
-	case "destinationPodNamespace":
-		assert.Equal(t, flow["DstK8S_Namespace"], element.GetStringValue())
-	case "destinationPodName":
-		assert.Equal(t, flow["DstK8S_Name"], element.GetStringValue())
-	case "sourceNodeName":
-		assert.Equal(t, flow["SrcK8S_HostName"], element.GetStringValue())
-	case "destinationNodeName":
-		assert.Equal(t, flow["DstK8S_HostName"], element.GetStringValue())
-	case "timeFlowRttNs":
-		assert.Equal(t, uint64(flow["TimeFlowRttNs"].(int64)), element.GetUnsigned64Value())
-	case "sourceMacAddress":
-	case "destinationMacAddress":
-		// Getting some discrepancies here, need to figure out why
-	default:
-		assert.Fail(t, "missing check on element", element.GetName())
+	name := element.GetName()
+	mapping, ok := write.MapIPFIXKeys[name]
```

Comment:
In the end, should this be configurable ?

I feel the hardcoded mapping makes it really specific to our usage

Reply:
True, it should read some of the mapping from the k8s rules config, but I wasn't up for doing this refactoring just for this little bug fix :)

Reply:
I don't believe we will ever *need* that in the operator, it's unlikely that we use IPFIX to get agent's flows imho; so I opened an upstream issue: https://github.com/netobserv/flowlogs-pipeline/issues/632